src/Ng/Jobs/Catalog2RegistrationJob.cs

```diff
@@ -78,12 +80,24 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
                 source,
                 storageFactories.LegacyStorageFactory);
 
-            RegistrationMakerCatalogItem.PackagePathProvider = new PackagesFolderPackagePathProvider();
+            if (isContentFlatContainer)
+            {
+                var flatContainerCursorUriString = arguments.GetOrThrow<string>(Arguments.FlatContainerCursorUri);
```

Comment:
You can very easily calculate the flat-container's cursor uri by using the flat-container's base address. You shouldn't need to pass it in via a command line argument.